userspace/falco/app/actions/load_rules_files.cpp

```diff
@@ -83,6 +84,9 @@ falco::app::run_result falco::app::actions::load_rules_files(falco::app::state&
 		{
 			falco_logger::log(falco_logger::level::WARNING,res->as_string(true, rc) + "\n");
 		}
+#if defined(__linux__) and !defined(MINIMAL_BUILD) and !defined(__EMSCRIPTEN__)
+		s.config->m_loaded_rules_filenames_sha256sum.push_back(falco::utils::calculate_file_sha256sum(filename));
```

Comment:
What if `m_loaded_rules_filenames_sha256sum` (and `m_loaded_configs_filenames_sha256sum`) was instead a `unoredered_set<string>string`, where key is filename and value is sha256sum? 
This would allow metrics and stats writer to just iterate over `m_loaded_rules_filenames_sha256sum` (and `m_loaded_configs_filenames`) keys.

Reply:
If that's more desirable, no problems to change it.

Reply:
I think it would simplify a bit the falco_metrics and stats_writer loops, making them more readable.

Reply:
Ok.